File: app/src/main/java/com/elseboot3909/GCRClient/API/AccountAPI.java
package com.elseboot3909.GCRClient.API;

import retrofit2.Call;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface AccountAPI {

    @GET("accounts/{user}")
    Call<String> getAccountInfo(@Path("user") String user);

    @GET("a/accounts/self/detail")
    Call<String> getSelfAccountDetails();

    @GET("a/accounts/self/starred.changes")
    Call<String> getStarredChanges();

    @PUT("a/accounts/self/starred.changes/{id}")
    Call<String> putStarredChange(@Path(value = "id", encoded = true) String id);

    @DELETE("a/accounts/self/starred.changes/{id}")
    Call<String> removeStarredChange(@Path(value = "id", encoded = true) String id);

}

File: app/src/main/java/com/elseboot3909/gcrclient/api/AccountAPI.kt
package com.elseboot3909.gcrclient.api

import retrofit2.Call
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.PUT
import retrofit2.http.Path

interface AccountAPI {

    @GET("accounts/{user}")
    fun getAccountInfo(@Path("user") user: String): Call<String>

    @GET("a/accounts/self/detail")
    fun getSelfAccountDetails(): Call<String>

    @GET("a/accounts/self/starred.changes")
    fun getStarredChanges(): Call<String>

    @PUT("a/accounts/self/starred.changes/{id}")
    fun putStarredChange(@Path(value = "id", encoded = true) id: String): Call<String>

    @DELETE("a/accounts/self/starred.changes/{id}")
    fun removeStarredChange(@Path(value = "id", encoded = true) id: String): Call<String>

}
